src/map/utils/fishingcontest.cpp

```diff
@@ -0,0 +1,716 @@
+
+            if (newStatus == CONTEST_STATUS_RELEASING)
+            {
+                ScoreContest();
+                UpdateRankingHistory();
+            }
+
+            if (newStatus == CONTEST_STATUS_CONTESTING)
+            {
+                InitNewContest();
+            }
+            else
+            {
+                SetContestStatus(newStatus);
+            }
+
+            SetContestChangeTime(GetStageChangeTime((FISHING_CONTEST_STATUS)newStatus, currentTime));
+
+            auto onStageProgress = lua["xi"]["fishingcontest"]["onStageProgress"];
+            if (!onStageProgress.valid())
+            {
+                sol::error err = onStageProgress;
+                ShowError("fishingcontest:onStageProgress: %s", err.what());
+                return;
+            }
+            auto result = onStageProgress(newStatus);
+            if (!result.valid())
+            {
+                sol::error err = result;
+                ShowError("fishingcontest:ProgressContest:(onStageProgress): %s", err.what());
+                return;
+            }
+        }
+    }
+
+    void UpdateRankingHistory()
+    {
+        // This should only be run once per contest, once the contest progresses from "Accepting" to "Releasing"
+        // Traverse the entries and update the database for each player
+        for (FishingContestEntry* it : FishingContestEntries)
+        {
+            // Get the Player ID
+            uint32 charID = 0;
+            auto   ret    = _sql->Query("SELECT charid FROM chars WHERE charname = '%s' LIMIT 1", it->name);
+            if (ret != SQL_ERROR && _sql->NumRows() != 0 && _sql->NextRow() == SQL_SUCCESS)
+            {
+                charID = _sql->GetUIntData(0);
+            }
+
+            if (charID > 0)
+            {
+                // Get the rank for column header
+                uint8 rankGroup = 0;
+                if (it->contestRank > 0 and it->contestRank <= 4)
+                {
+                    rankGroup = it->contestRank;
+                }
+                else if (it->contestRank > 4 and it->contestRank <= 10)
+                {
+                    rankGroup = it->contestRank;
+                }
+
+                // Set the rank value
+                if (rankGroup > 0)
+                {
+                    const char* Query = "INSERT INTO char_fishing (charid, contest_rank_%u) VALUES (%u, 1) ON DUPLICATE KEY UPDATE contest_rank_%u = contest_rank_%u + 1;";
+                    int32       ret   = _sql->Query(Query, rankGroup, charID, rankGroup, rankGroup);
+
+                    if (ret != SQL_SUCCESS)
+                    {
+                        ShowWarning("Unable to update player [%s] fishing reward history.", it->name);
+                    }
+                }
+            }
+        }
+    }
+
+    void ScoreContest()
+    {
+        TracyZoneScoped;
+
+        // Sort the list first
+        // clang-format off
```

Comment:
Good try! you've wrapped the lambda correctly, but the formatting inside is wrong 😆 

It's truly a shame that clang-format hasn't made this possible yet